### ISSUE TITLE
Add Python 2 Support

### ISSUE THREAD

**Issue by brthor** [llllllllll/codetransformer#62]
I've already made good progress here. I'm quite aware of the challenges since python3 enhanced the capabilities of the dis module. Several modules need to be manually backported.

Reintegrating to retain python 3 support as well will require some effort.

I'm making this issue to judge interest in accepting a PR with this, otherwise I will maintain a separate fork.

**Comment by brthor** [comment]
@llllllllll @ssanderson

**Comment by llllllllll** [comment]
Sorry about the slow response, unfortunately, we would not accept a PR to add Python 2 support to codetransformer.

Codetransformer is developed for fun, and maintaining Python 2/3 compatibility is frustrating. We like using new Python features that were added in Python 3 and don't want to deal with the incompatibilities. We find it is subjectively less enjoyable to develop Python 2 compatible code.

The best path forward would be a fork under a compatible license, maybe named `codetransformer-py2`.

Based on your github it looks like it might be for https://github.com/brthor/lambdazen/blob/master/HowItWorks.md#compiled-bytecode-rewriting, is that the case?

**Comment by brthor** [comment]
Thanks for your response!

I don't disagree that python2 vs python3 compatibility is terrible to maintain. I am not planning to reintegrate python3 support for that reason, and will likely continue to use your original version for python3.

I figured this would be the case, seeing as how widely used python3 features were used in the codebase. No worries, `codetransformer-py2` sounds quite reasonable, and I think I'll put a note on why the separation on the readme. 

Indeed, I was exploring bytecode rewriting for that project when I came across codetransformer. Recently, I've had some experiments using it that need python2 support. 

I completed the backport, with passing tests (except the decompiler), which mainly required backporting parts of the 3.4 version of the `dis` std library to 2.7 (for any interested parties). I plan to publish it soon.
